### PR TITLE
jak2: add missing pc options to progress menu

### LINKED ISSUE #3116: Left/Right (3rd Person) is Inverted Permanently
### Acknowledgements

- [X] I have [searched](https://github.com/open-goal/jak-project/issues?q=is%3Aissue+is%3Aopen+label%3Abug+label%3Ajak2+sort%3Aupdated-desc+) for duplicate or closed bug reports
- [X] I understand that I am supposed to provide my own legitimately obtained copy of the game

### Describe the Bug

It doesn't matter if you set Left/Right (3rd Person) camera controls to Normal or Inverted, they act inverted at all times.

### How To Reproduce

It doesn't matter if you set Left/Right (3rd Person) camera controls to Normal or Inverted, they act inverted at all times.

This was done using an NTSC ISO, if that's relevant.

### Does this problem occur on original hardware or PCSX2?

Yes, it's unique to OpenGOAL

### Expected Behavior

Normal should have Left be Left and Right be Right.

### Environment Information

[opengoal-support-package.zip](https://github.com/open-goal/jak-project/files/13239508/opengoal-support-package.zip)

### Game Version

NTSC 1.0 (black label)

### Have you set the game to something other than `60fps`?

No

### PR DESCRIPTION
**Pull request by Hat-Kid**
Adds the majority of the missing PC options to the progress menu, such as most of the graphics options, camera options and a quit game button on the title screen menu.

Fixes #3116 

image
image

### CODE REVIEW

File: goal_src/jak2/pc/progress/progress-static-pc.gc [hunk 1]
```diff
@@ -310,40 +340,267 @@ This gives us more freedom to write code how we want.
...
+          (commit-to-file *pc-settings*)))
+      (new 'static 'menu-generic-boolean-option
+        :name (text-id progress-graphics-actor-culling)
+        :truthy-text (text-id progress-on)
+        :falsey-text (text-id progress-off)
+        :get-value-fn (lambda () (-> *pc-settings* ps2-actor-vis?))
+        :should-disable? (lambda () (-> *pc-settings* speedrunner-mode?))
+        :on-confirm (lambda ((val symbol))
+          (set! (-> *pc-settings* ps2-actor-vis?) val)
+          (commit-to-file *pc-settings*)))
```
Comment (ManDude) [inline goal_src/jak2/pc/progress/progress-static-pc.gc:424]: Do not add this option. It's not a good idea for Jak 2.

File: goal_src/jak2/pc/progress/progress-static-pc.gc [hunk 2]
```diff
@@ -310,40 +340,267 @@ This gives us more freedom to write code how we want.
...
+          (set! (-> *pc-settings* ps2-parts?) val)
+          (commit-to-file *pc-settings*)))
+      (new 'static 'menu-generic-boolean-option
+        :name (text-id progress-graphics-force-envmap)
+        :truthy-text (text-id progress-on)
+        :falsey-text (text-id progress-off)
+        :get-value-fn (lambda () (-> *pc-settings* force-envmap?))
+        :on-confirm (lambda ((val symbol))
+          (set! (-> *pc-settings* force-envmap?) val)
+          (commit-to-file *pc-settings*)))
```
Comment (ManDude) [inline goal_src/jak2/pc/progress/progress-static-pc.gc:440]: Don't add this one either.

**Review by ManDude — CHANGES_REQUESTED** [review]

File: goal_src/jak2/pc/progress/progress-static-pc.gc [hunk 3]
```diff
@@ -310,40 +340,250 @@ This gives us more freedom to write code how we want.
...
+            (set! (-> *pc-settings* fast-elevator?) val)
+            (commit-to-file *pc-settings*)))
+        (new 'static 'menu-generic-boolean-option
+          :name (text-id progress-fast-progress)
+          :truthy-text (text-id progress-on)
+          :falsey-text (text-id progress-off)
+          :get-value-fn (lambda () (-> *pc-settings* fast-progress?))
+          :on-confirm (lambda ((val symbol))
+            (set! (-> *pc-settings* fast-progress?) val)
+            (commit-to-file *pc-settings*))))))
```
Comment (ManDude) [inline goal_src/jak2/pc/progress/progress-static-pc.gc:367]: Can you actually move these to only show up for game options in the title screen? I don't want to deal with potential bugs of changing this mid-elevator and stuff (not that I believe there are any, but yeah).

**Review by ManDude — CHANGES_REQUESTED** [review]

**Review by xTVaser — APPROVED** [review]
Pulled it down and tested the major things -- if graphics options save / if the camera changes work.  They did

**Review by ManDude — APPROVED** [review]

**Pull request merged**